Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -155,11 +165,11 @@ public BlackboardArtifactNode(BlackboardArtifact artifact) {
         this.setName(Long.toString(artifact.getArtifactID()));
         this.setDisplayName();
         this.setIconBaseWithExtension(ExtractedContent.getIconFilePath(artifact.getArtifactTypeID())); //NON-NLS
-        Case.addPropertyChangeListener(pcl);
+        Case.addEventSubscriber(CASE_EVENTS_OF_INTEREST, WeakListeners.propertyChange(pcl, null));
```

Comment:
Yeah, my initial attempt involved modifying LocalEventPublisher to wrap everything in a weak reference but that resulted in some events no longer being processed. Presumably the listener(s) in question were not being held onto by a strong reference elsewhere so they were getting garbage collected. I suspect using lambdas as a listener might have a role in this somehow but don't have the time at this point to continue investigation.